src/syscheckd/syscheck_audit.c

```diff
@@ -434,7 +451,9 @@ int audit_init(void) {
 
     // Check if auditd is installed and running.
     int aupid = check_auditd_enabled();
-    if (aupid <= 0) {
+    int audit_status = audit_get_mode();
+
+    if (aupid <= 0 || audit_status < 1) {
         mwarn(FIM_AUDIT_NORUNNING);
         return (-1);
     }
```

Comment:
The message printed by this `mwarn` states that auditd is not running, could it happen that auditd is running and audit is disabled? Does it make sense to use a different log message for this situation.

Also, `audit_get_mode` returns a value from the newly created `audit_status` but here it's being casted to an int, can we change the type of storage variable and change the if condition to properly reflect this?